source/iopipe/json/serialize.d

```diff
@@ -31,6 +31,95 @@ import std.typecons : Nullable;
 import std.conv;
 import std.format;
 
+struct DefaultDeserializationPolicy {
+    ReleasePolicy relPol = ReleasePolicy.afterMembers; // default policy
+        
+    // Called at beginning of struct/class deserialization
+    bool[SerializableMembers!T.length] onBegin(JT, T)(ref JT tokenizer) {
```

Comment:
I'm thinking this should be named `onObjectBegin`, to denote the thing we are parsing. We will need an `onArrayBegin` too when we get to arrays/ranges.

I think the names should follow the JSON spec, and let the policy decide how to handle each piece.

This is looking more and more like a SAX-style parser, but with compile-time context of the type being deserialized. It feels like a good API!